src/windows/system.rs

```diff
@@ -227,6 +228,9 @@ impl SystemExt for System {
 
         loop {
             let mut cb_needed = 0;
+            // reserve(n) ensures the Vec has capacity for n additional elements
+            // so we should reserve buffer_size - len
+            process_information.reserve(buffer_size - process_information.len());
```

Comment:
Unless I missed something, `process_information`'s length is always 0 until we use `set_len` below. So why subtracting the length?

Reply:
That is true, but it would be pretty easy for a future change to this function to allow reaching this point with a length set as it's pretty long and hard to follow.

I'd feel more comfortable removing the len part if this function broke out of the loop immediately before the set_len. Would you be okay with a refactor like that?

Reply:
Sure, go ahead!